core/src/main/java/org/infinispan/distribution/ch/KeyPartitioner.java

```diff
@@ -25,6 +25,10 @@ default void init(HashConfiguration configuration) {
       // Do nothing
    }
 
+   default void init(KeyPartitioner other) {
```

Comment:
The way I imagined the KeyPartitioner interface, it would only be initialized with the HashConfiguration when the cache is wired, and there shouldn't be anything worth copying from another instance.

I'm really curious why you thought `init(HashConfiguration)` wouldn't be enough and you had to call `init(KeyPartitioner)` as well.